combat-trainer.lic

```diff
@@ -1954,19 +1965,27 @@ class AbilityProcess
     @khri
       .map { |name| "Khri #{name}" }
       .each do |name|
-        next if game_state.danger && kneel_for_khri?(@kneel_khri, name)
+        next if game_state.danger && DRCA.kneel_for_khri?(@kneel_khri, name)
         timer = game_state.cooldown_timers[name]
         next unless !timer || (Time.now - timer).to_i > 30
-        game_state.cooldown_timers[name] = Time.now if activate_khri?(@kneel_khri, name)
+        game_state.cooldown_timers[name] = Time.now if DRCA.activate_khri?(@kneel_khri, name)
       end
     @barb_buffs
       .select { |name| Flags["ap-#{name}-expired"] }
       .each do |name|
-        Flags.reset("ap-#{name}-expired")
-        activate_barb_buff?(name)
-        echo "Activated barb_buff #{name}!" if $debug_mode_ct
-        waitrt?
-        pause 6 if get_data('spells').barb_abilities[name]['type'].eql? 'meditation'
+        if DRStats.mana < @barb_buffs_inner_fire_threshold
+          echo "Not activating barb_buff #{name} until have at least #{@barb_buffs_inner_fire_threshold} inner fire. You currently have #{DRStats.mana}." if $debug_mode_ct
```

Comment:
Important: don't try to activate an ability unless you have sufficient inner fire in reserve. If you don't have enough then your active abilities will drop and that's no bueno.